libs/ui-lib/lib/cim/components/Agent/tableUtils.tsx

```diff
@@ -343,6 +343,10 @@ export const canUnbindAgent = (
     }
   }
 
+  if (!agent.metadata?.labels?.hasOwnProperty(AGENT_BMH_NAME_LABEL_KEY)) {
```

Comment:
it looks like this will also disallow removing agents from CIM-flow cluster ?

We need to ensure this gets applied only on AI-flow cluster. To do that you need to get the `InfraEnv` of `Agent` and check if `InfraEnv.spec.clusterRef` is defined.